ytypes/leaf.go

```diff
@@ -216,17 +216,27 @@ func validateUnion(schema *yang.Entry, value interface{}) util.Errors {
 	}
 
 	util.DbgPrint("validateUnion %s", schema.Name)
-	// Must be a ptr - either a struct ptr or Go value ptr like *string.
-	// Enum types are also represented as a struct for union where the field
-	// has the enum type.
-	if reflect.TypeOf(value).Kind() != reflect.Ptr {
-		return util.NewErrs(fmt.Errorf("wrong value type for union %s: got: %T, expect ptr", schema.Name, value))
+	rkind := reflect.TypeOf(value).Kind()
+	var v reflect.Value
```

Comment:
maybe tighten it up with:

v := reflect.ValueOf(value)
switch v.Type().Kind() {
case reflect.Ptr:
  v = v.Elem()
case reflect.Int64:
default:
  return util.NewErrs(...)
}